daffodil-core/src/main/scala/org/apache/daffodil/dpath/Expression.scala

```diff
@@ -2135,6 +2143,43 @@ case class FNTwoArgsMathExpr(nameAsParsed: String, fnQName: RefQName,
   }
 }
 
+case class DFDLXShiftExpr(nameAsParsed: String, fnQName: RefQName,
+                             args: List[Expression], constructor: (List[CompiledDPath], NodeInfo.Kind) => RecipeOp)
+  extends FunctionCallBase(nameAsParsed, fnQName, args) {
+  checkArgCount(2)
+  override lazy val inherentType = {
+
+    val argInherentType = args(0).inherentType
+    schemaDefinitionUnless(
+      argInherentType.isSubtypeOf(NodeInfo.PrimType.UnsignedLong) ||
+        argInherentType.isSubtypeOf(NodeInfo.PrimType.Long),
+      "First argument must be either xs:unsignedLong or xs:long or a subtype of those, but was %s.",
+      argInherentType)
```

Comment:
I'm not positive but I think this should be `argInherentType.globalQName` in the error message? I think otherwise you can the scala name for the class, e.g. `Integer`, where we really want `xs:integer`.